### DISCUSSION TITLE
MZI with Trench and Circular Bend

### DISCUSSION THREAD

**Discussion by sheronlian** [gdsfactory/gdsfactory#2186] [Q&A] [answered]
```
r = 80
wg_width=0.5
mmi2x1_width = 6
mmi2x1_length = 30
taper_width = 1.5
taper_length = 20
mmi2x1_wg_sep = 3.02/2

strip = gf.cross_section.cross_section(cladding_layers=[(3, 0)], cladding_offsets=[3])
bend = gf.components.bend_circular(radius = r, cross_section = strip)
straight = gf.components.straight(cross_section = strip)
mmi1x2 = gf.components.mmi1x2(width = wg_width,width_taper=taper_width,length_taper=taper_length,length_mmi=mmi2x1_length,width_mmi=mmi2x1_width,gap_mmi=mmi2x1_wg_sep, cross_section = strip)
mzi = gf.components.mzi(delta_length=0, length_y=150-r, splitter=mmi1x2, bend = bend)

```

I am trying to generate a MZI with trench for positive resist and a circular bend instead of the default Euler bend. The above code would work if I use the default Euler bend but when I switch the bend to circular bend, this code would give an error: 

ValueError: Cannot apply kwargs {'cross_section': 'strip'} to 'bend_circular_066b11ba' 

<img width="1316" alt="Screenshot 2023-10-15 at 11 45 57 PM" src="https://github.com/gdsfactory/gdsfactory/assets/148023460/78fdc5b0-f4dc-4bc8-9f6f-2cb7a09f0cc9">
I am also seeing sharp cladding around the MMI squares. Is there a way to round cladding edges? Thank you so much!

**Comment by sheronlian** [comment]
Update: I fixed the problem using partial. However it would still be nice if there is a way to round the corners around the MMIs like the image below. My initial idea would be change the cross section around the tapers... if there's an easier way that would also be helpful, thanks!
<img width="1017" alt="Screenshot 2023-10-16 at 10 24 44 AM" src="https://github.com/gdsfactory/gdsfactory/assets/148023460/c79d2843-3915-4a93-bcdb-0856f500846e">

**Comment by wwwwwwzx** [comment]
try the `get_polygon_enclosure` method and add the resulted polygon to the mmi

**Comment by joamatab** [comment, marked as answer]
I think this is what you want do to
```python

from functools import partial 
import gdsfactory as gf

r = 80
wg_width = 0.5
mmi2x1_width = 6
mmi2x1_length = 30
taper_width = 1.5
taper_length = 20
mmi2x1_wg_sep = 3.02 / 2

strip = partial(gf.cross_section.cross_section,cladding_layers=[(3, 0)], cladding_offsets=[3])
bend = gf.components.bend_circular(radius=r, cross_section=strip)
straight = gf.components.straight(cross_section=strip)
mmi1x2 = gf.components.mmi1x2(
    width=wg_width,
    width_taper=taper_width,
    length_taper=taper_length,
    length_mmi=mmi2x1_length,
    width_mmi=mmi2x1_width,
    gap_mmi=mmi2x1_wg_sep,
    cross_section=strip,
)
mzi = gf.components.mzi(
    delta_length=0,
    length_y=150 - r,
    splitter=mmi1x2,
    bend=bend,
    straight=straight,
    cross_section=strip,
)
mzi.show()

```

![image](https://github.com/gdsfactory/gdsfactory/assets/4514346/8f8253d7-1f97-45d3-b791-80d79b83c089)
